File: 2024/May/4-1 2000 Reverse Prefix of Word/ReversePrefix.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        index = 0
        result = word
        for index in range(len(word)):
            if word[index] != ch:
                continue
            
            result = ''
            for i in range(index, -1, -1):
                result += word[i]
            for index in range(index + 1, len(word), 1):
                result += word[index]
            break

        return result

File: 2024/May/4-1 2000 Reverse Prefix of Word/ReversePrefix.java
class Solution {
    public String reversePrefix(String word, char ch) {
        char[] arr = word.toCharArray();
        int left = 0;
        int right;
        for(right = 0; right < word.length(); right++) {
            if(arr[right] != ch) {
                continue;
            }

            char temp;
            while(left < right) {
                temp = arr[left];
                arr[left] = arr[right];
                arr[right] = temp;
                left++;
                right--;
            }
            break;
        }

        return new String(arr);
    }
}